lib/linter/apply-disable-directives.js

```diff
@@ -82,17 +201,22 @@ function applyDirectives(options) {
         }
     }
 
-    const unusedDisableDirectives = options.directives
-        .filter(directive => directive.type === "disable" && !usedDisableDirectives.has(directive))
-        .map(directive => ({
+    const unusedDisableDirectivesToReport = options.directives
+        .filter(directive => directive.type === "disable" && !usedDisableDirectives.has(directive));
+
+    const processed = processUnusedDisableDirectives(unusedDisableDirectivesToReport);
+
+    const unusedDisableDirectives = processed
+        .map(({ description, fix, position }) => ({
             ruleId: null,
-            message: directive.ruleId
-                ? `Unused eslint-disable directive (no problems were reported from '${directive.ruleId}').`
+            message: description
+                ? `Unused eslint-disable directive (no problems were reported from ${description}).`
                 : "Unused eslint-disable directive (no problems were reported).",
-            line: directive.unprocessedDirective.line,
-            column: directive.unprocessedDirective.column,
+            line: position.line,
+            column: position.column,
             severity: options.reportUnusedDisableDirectives === "warn" ? 1 : 2,
-            nodeType: null
+            nodeType: null,
+            fix: options.disableFixes ? null : fix
```

Comment:
```suggestion
            ...options.disableFixes ? {} : { fix }
```

`null` isn't an allowed value of the `fix` property, the property should be omitted if no fix is available.